public/population_submission/population_submission.rb

```diff
@@ -72,57 +72,64 @@ def test_error(response)
-        response = call_bip request
-        if response['meta']['total_count'] == 0
-          create_record('plant_variety',
-          plant_variety_name: row[VARIETY],
-          crop_type: row[CROP_TYPE])
-        
-        else
-          response['plant_varieties'][0]['id']
-        end
-      end
+def record_plant_variety(plant_variety_name, crop_type)
+  request = Net::HTTP::Get.new("/api/v1/plant_varieties?plant_variety[query][plant_variety_name]=#{URI.escape plant_variety_name}", @headers)
+  response = call_bip request
+  if response['meta']['total_count'] == 0
+    create_record('plant_variety',
+      plant_variety_name: plant_variety_name,
+      crop_type: crop_type
+    )
+  else
+    response['plant_varieties'][0]['id']
+  end
+end
 
 
 puts ' 3. Submitting plant_lines'
 
-      def record_plant_lines(plant_line_name, plant_variety_name)
-        request = Net::HTTP::Get.new("/api/v1/plant_linesplant_line[query][plant_line_name]=#{URI.escape plant_line_name}", @headers)
-        response = call_bip request
-        if response['meta']['total_count'] == 0
-          create_record('plant_line',
-          Plant_line_name: row[LINE_NAME],
-          plant_variety_name: row[VARIETY])
-        else
-          response['plant_lines'][0]['id']
-        end
-      end
+def record_plant_line(plant_line_name, plant_variety_id)
+  request = Net::HTTP::Get.new("/api/v1/plant_lines?plant_line[query][plant_line_name]=#{URI.escape plant_line_name}", @headers)
+  response = call_bip request
+  if response['meta']['total_count'] == 0
+    create_record('plant_line',
+      plant_line_name: plant_line_name,
+      plant_variety_id: plant_variety_id
+    )
+  else
+    response['plant_lines'][0]['id']
+  end
+end
+
+def associate_line_with_population(plant_line_id, plant_population_id)
+  create_record('plant_population_list',
+    plant_line_id: plant_line_id,
+    plant_population_id: plant_population_id
+  )
+end
 
 puts '4. Submitting plant_accessions'
 
-      def record_plant_accessions(plant_accession, accession_originator,plant_variety_id, comments)
+      def record_plant_accessions(plant_accession, originating_organisation,plant_line_id,comments)
         request = Net::HTTP::Get.new("/api/v1/plant_accessions?plant_accession[query][plant_accession]=#{plant_accession}", @headers)
         response = call_bip request
         plant_accession_id = if response['meta']['total_count'] == 0
-                      plant_variety_id = record_plant_variety(row[VARIETY], row[CROP_TYPE])
                         create_record('plant_accession',
-                            plant_accession: row[ACCESSION_NAME],
-                            originating_organisation: row[ACCESSION_SOURCE],
+                            plant_accession: plant_accession,
+                            originating_organisation: originating_organisation,
                             plant_variety_id: plant_variety_id,
```

Comment:
You probably don't want to do this. I think it is enough that a PV is related to a PL, since that PL is further related to a PV.